ui/src/app/views/app/component.html

```diff
@@ -1,55 +1,17 @@
 <ng-template *ngIf="(user | async).isSignedIn; then signedIn else notSignedIn"></ng-template>
-<ng-template #logoImages>
-  <a class="image-link-wrapper" routerLink="/">
-    <img class="main-logo" src="assets/images/all-of-us-logo.svg" alt="All of Us"
-    style="height: 100%;">
-  </a>
-  <img class="portal-logo" src="assets/images/portal-logo.svg" alt="Researcher Portal"
-    style="margin-left: 0.5rem; height: 100%;">
-</ng-template>
-<ng-template #notSignedIn>
 
-  <div *ngIf="currentUrl && currentUrl.indexOf('/data-browser/') === 0">
-    <div class="page-container">
-      <div style="height: 3rem;">
-        <ng-template *ngIf="true; then logoImages"></ng-template>
-      </div>
-      <div>
-        <button (click)="signIn()" class="btn">Sign In to Workbench</button>
-      </div>
-    </div>
-    <div>
-      <!--
-        Angular's builtin Router module will attach specific views here
-        based on clicking routerLink elements (above) or router.navigate calls.
-      -->
-      <router-outlet></router-outlet>
-    </div>
-  </div>
+<ng-template #notSignedIn>
 
   <div *ngIf="currentUrl && currentUrl.indexOf('/data-browser/') === -1" class="sign-in-container">
-    <div class="sign-in-inner-container" [style.margin-top]="getTopMargin()">
-      <div style="height: 2rem; margin: 0.5rem 2rem;">
-        <ng-template *ngIf="true; then logoImages"></ng-template>
-      </div>
-      <div *ngIf="!showCreateAccount;" style="margin-top: 1.5rem;">
-        <div style="display: flex; justify-content: space-around;">
-          <button (click)="showCreateAccount = true" class="btn btn-primary">Create Account</button>
-        </div>
-        <div style="display: flex; justify-content: space-around;">
-          <button (click)="signIn()" class="btn btn-primary">Sign In</button>
-        </div>
-      </div>
-      <div *ngIf="showCreateAccount;">
-        <app-invitation-key></app-invitation-key>
-      </div>
-    </div>
+    <app-page-template-signed-out style="width: 100%" [imageSrc]=backgroundImgSrc>
```

Comment:
May I suggest moving the wrapper into the template for the invitation key page?  Such that in `invitation-key/component.html` you would have

```
<app-page-template-signed-out [imageSrc]="invKeyBackgroundImgSrc">
  /* The existing invitation key template */
</app-page-template-signed-out>
```

Then each page can define its own background image and pass that through to the scaffolding w/out the top level app component really needing to know anything what image it wants as its background.

This would also allow you to route directly to this page if you so desired (though that's probably out of scope for this PR).

Reply:
Yeah i had it inside the invitation page but then moved it out to App as i thought thats the starting point and invitation code, like other pages should just dictate the image source. If i move it under invitation key all the following pages will be injection InvitationKeyComponent? or is there any other alternative that i am missing?

Reply:
If you have three components, say `invitation-key`, `create-account`, `terms-of-service` (I'm making up some names here obviously) then what I'm suggesting would be to have:

```
// In invitation-key/component.html
<app-page-template-signed-out [imageSrc]="invKeyBackgroundImgSrc">
  /* The existing invitation key template */
</app-page-template-signed-out>

// In create-account/component.html
<app-page-template-signed-out [imageSrc]="createAccountBackgroundImgSrc">
  /* The rest of the create-account component template */
</app-page-template-signed-out>

// In terms-of-service/component.html
<app-page-template-signed-out [imageSrc]="tosBackgroundImgSrc">
  /* The rest of the terms-of-service html*/
</app-page-template-signed-out>
```

Then the template will provide some context and scaffolding for each component but can also be parametrized _by that component_.  Then the app component (or a router) can do logic to determine which component to display without needing to know about its specific needs w.r.t background images (or whatever else may need to be abstracted into the page template).

Does that make sense?

c.c. @blrubenstein

Reply:
Unfortunately, the create-account element is actually contained within the invitation-key component, because we need to gate access through invitation key (as it stands right now). That's actually a large part of why we needed to figure out how to change the background image like this. So in several ways, its actually the invitation key component that has been deciding which to display. (And then the app component is handling structure and display.)

Reply:
Oh!  I was under the (wrong) impression that each step in the registration process was modeled by a component, and that all those components were _siblings_.  But if one is a child of another, then you'd wind up with the page template twice, which won't work.  So you probably either want to break the "contains" relationship by passing that information through a service, local storage, or route param; or re-evaluate if my original suggestion meets your needs (it may very well not).

I guess I never checked my assumption that these components were siblings, my bad :/

Reply:
Not a problem. Yeah, its a little confusing because of how we have been handling the registration piece. I think the current method proposed here is fine, because I don't expect to see more pieces needing to be passed up to the page template.